graylog2-web-interface/src/components/configurations/UrlWhiteListForm.jsx

```diff
@@ -0,0 +1,204 @@
+      return (
+      // eslint-disable-next-line react/no-array-index-key
+        <tr key={url.id}>
+          <td>{idx + 1}</td>
+          <td>
+            <Input type="text"
+                   id={`title-input${idx}`}
+                   ref={(elem) => { inputs[`title${idx}`] = elem; }}
+                   help={validationState.errors[idx] && validationState.errors[idx].title && !validationState.errors[idx].title.valid ? 'required' : null}
+                   name="title"
+                   bsStyle={validationState.errors[idx] && validationState.errors[idx].title && !validationState.errors[idx].title.valid ? 'error' : null}
+                   onChange={event => _onInputChange(event, idx, url.type)}
+                   defaultValue={url.title}
+                   required />
+          </td>
+          <td>
+            <Input type="text"
+                   id={`value-input${idx}`}
+                   ref={(elem) => { inputs[`value${idx}`] = elem; }}
+                   help={validationState.errors[idx] && validationState.errors[idx].value && !validationState.errors[idx].value.valid ? 'Not valid' : null}
+                   name="value"
+                   bsStyle={validationState.errors[idx] && validationState.errors[idx].value && !validationState.errors[idx].value.valid ? 'error' : null}
+                   onChange={event => _onInputChange(event, idx, url.type)}
+                   defaultValue={url.value}
+                   required />
+          </td>
+          <td>
+            <Input id={`url-input-type-${idx}`}
+                   required
+                   autoFocus>
+              <Select placeholder="Select Cache Type"
+                      clearable={false}
+                      options={options}
+                      matchProp="label"
+                      onChange={option => _onUpdateType(idx, option)}
+                      value={url.type} />
+            </Input>
+          </td>
+          <td>
+            <span className="">
+              <Icon name="fa-trash" style={{ cursor: 'pointer' }} onClick={event => _onRemove(event, idx)} />
+            </span>
+          </td>
+        </tr>
+      );
+    }));
+  };
+
+  useEffect(() => {
+    const valid = _isFormValid();
+    update(state, valid);
+  }, [state]);
+
+  return (
+    <>
+      <Input type="checkbox"
+             id="whitelist-disabled"
+             label="Disabled"
+             checked={state.disabled}
+             onChange={() => setState({ ...state, disabled: !state.disabled })}
+             help="Disable this white list." />
+      <Table striped bordered className="top-margin">
+        <thead>
+          <tr>
+            <th>#</th>
+            <th>Title</th>
+            <th>Url</th>
+            <th>Type</th>
+            <th>Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {_summary()}
+        </tbody>
+      </Table>
+      <Button bsSize="xs" onClick={event => _onAdd(event)}>Add Url</Button>
```

Comment:
I think this button is a bit small and should have a regular size. We should also consider adding one on top of the table, ensuring it will be reachable even if there are many items in the whitelist.

Reply:
Updated the button to regular size and added it on the top